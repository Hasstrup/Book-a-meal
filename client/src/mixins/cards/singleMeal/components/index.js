import React from 'react';
import { changePhoto, HideMealCard } from '../../../../modules/WorkStation/utils';
import { AddToCart } from '../../../../actions/orders/';


const defaultUrl = 'https://images.unsplash.com/photo-1453831362806-3d5577f014a4?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=9dd8da96be0724ab84e4147d428f6bba&auto=format&fit=crop&w=500&q=60';
/**
 * @name SingleMealCard
 * @returns {function} React Component
 * @param {Object} Props The props passed to the item
 * wshould contain the current kitchen the item and the item;
 * @description Single Meal Card component, recieves data from it's parent container
 */
const SingleMealCard = ({ meal, kitchen, handleDelete, handleEdit, displayText, wantsToEdit, renderEditable, dispatch }) => (
    <div className="display-menu-item" id={meal.id} onMouseLeave={() => { HideMealCard(meal.id) }}>
      <div className="order-add-item">
        <img src={meal.image || defaultUrl} alt={meal.name} id={`image-${meal.id}`} className="main-meal-item-img" />
        <div className={`change-preview-image modifiable${meal.id}`} style={{ opacity: wantsToEdit ? 1 : 0, zIndex: wantsToEdit ? 5 : -1, display: wantsToEdit ? 'block' : 'none' }}>
          <p className="label-for-input"> Click to change Image </p>
          <input type="file" className={`input-file image-${meal.id}`} name="input-file" onChange={() => { changePhoto(`image-${meal.id}`)}} accept=".jpg, .png, .jpeg" />
        </div>
      </div>
      <div className="item-details">
        <div className="item-title-and-description">
          <p className={`item-title-main editable editable${meal.id}`}> {meal.name || ''} </p>
          <p className={`item-description editable editable${meal.id}`}> {meal.description || ''} </p>
        </div>
        <div className="buttons-array-and-togglers">
          {
          kitchen && kitchen.id === meal.kitchenId ?
          <p className="option-options"> { renderEditable && <span className="edit-meal-option" onClick={handleEdit} id={`target${meal.id}`}> {displayText} </span> } { wantsToEdit ? null : <span className="delete-meal-option" onClick={handleDelete}> Delete </span> } </p>
          : <p className="option-options" onClick={() => dispatch(AddToCart()(meal))}> Order </p>
        }
  
          <p className="option-options2" > N{meal.price} </p>
        </div>
      </div>
    </div>
);

export default SingleMealCard;