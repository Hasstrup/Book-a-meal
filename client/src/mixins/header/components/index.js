import React from 'react';
import { LogOutUser } from '../../../actions/users/'

/**
   * @name LoggedOutHeader
   * @description This header would only be rendered when there is no user in session
   * @param {object} props the props containing the history object for navigation
   * @returns {function} React Component
   */
export const LoggedOutHeader = ({ history }) => (
  <header>
    <div className="header">
      <div className="home-tag">
        <p className="logo" onClick={() => { history.push('/'); }}> Book <br />
            a meal
        </p>
      </div>
      <div className="header-buttons">
        <p className="h1-login-button"> Login </p>
        <p> Sign Up</p>
        <p className="h1-login-button" onClick={() => { history.push('/catalogue'); }}> Catalogue </p>
      </div>
    </div>
  </header>
);


  /**
   * @name LoggedInHeader
   * @description This header would only be rendered when a user is logged in
   * @param {object} props the props containing the history object for navigation
   * @returns {function} React Component
   */
export const LoggedInHeader = ({ history }) => (
  <header>
    <div className="header">
      <div className="home-tag">
        <p className="logo" onClick={() => { history.push('/'); }}> Book <br />
            a meal
          </p>
      </div>
      <div className="header-buttons">
        <p className="h1-login-button" onClick={() => { history.push('/orders')}}> Cart & Orders </p>
        <p onClick={() => { history.push('/profile')}} > Workstation </p>
        <p className="h1-login-button" onClick={() => { history.push('/catalogue')}}> Catalogue </p>
        <p className="h1-login-button" onClick={() => { LogOutUser(history) }}> Logout </p>
      </div>
    </div>
  </header>
);
