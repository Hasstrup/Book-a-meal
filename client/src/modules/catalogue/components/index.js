import { CatalogueFirstRow, CatalogueGridMainX } from '../index';

export const ComponentList = [CatalogueFirstRow, CatalogueGridMainX];
export const PropList = [{}, { menus: [] }];
export const NameList = ['Catalogue Preamble', 'Catalogue Grid'];

