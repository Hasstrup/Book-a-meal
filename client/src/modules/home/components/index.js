import AuthComponent from './auth';
import BannerComponent from './banner';
import FloatingComponent from './floater';

export const ComponentList = [AuthComponent, BannerComponent, FloatingComponent];
export const PropList = [{ logInUser: () => {}, createUser: () => {} }, {}, {}];
export const NameList = ['Auth Component', 'Banner Component', 'Floating Component'];