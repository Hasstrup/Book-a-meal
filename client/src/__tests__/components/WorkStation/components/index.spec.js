import TestComponent from '../../../TestComponents';
import { ComponentList, PropList, NameList } from '../../../../modules/WorkStation/components';

(async () => {
  await TestComponent(ComponentList, NameList, PropList, 'WorkStation Components');
})();