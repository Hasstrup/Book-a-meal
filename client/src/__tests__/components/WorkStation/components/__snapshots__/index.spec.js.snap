// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkStation Components MenuOfTheDayContainer should render correctly on shallow render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuOfTheDayContainer
    dispatch={[Function]}
    meals={Array []}
    ofTheDay={
      Object {
        "description": "",
        "meals": Array [],
        "name": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          buttonText="Change"
          handleClick={[Function]}
          ofTheDay={
            Object {
              "description": "",
              "meals": Array [],
              "name": "",
            }
          }
          openModal={[Function]}
          previewImage="https://images.unsplash.com/photo-1482049016688-2d3e1b311543?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=c75e0437e819afdceeb3050a6bcdd71b&auto=format&fit=crop&w=653&q=80"
        />,
        <SelectMealModal
          closeModal={[Function]}
          handleSelect={[Function]}
          meals={Array []}
          selectedMeals={Array []}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonText": "Change",
          "handleClick": [Function],
          "ofTheDay": Object {
            "description": "",
            "meals": Array [],
            "name": "",
          },
          "openModal": [Function],
          "previewImage": "https://images.unsplash.com/photo-1482049016688-2d3e1b311543?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=c75e0437e819afdceeb3050a6bcdd71b&auto=format&fit=crop&w=653&q=80",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "closeModal": [Function],
          "handleSelect": [Function],
          "isVisible": undefined,
          "meals": Array [],
          "selectedMeals": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            buttonText="Change"
            handleClick={[Function]}
            ofTheDay={
              Object {
                "description": "",
                "meals": Array [],
                "name": "",
              }
            }
            openModal={[Function]}
            previewImage="https://images.unsplash.com/photo-1482049016688-2d3e1b311543?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=c75e0437e819afdceeb3050a6bcdd71b&auto=format&fit=crop&w=653&q=80"
          />,
          <SelectMealModal
            closeModal={[Function]}
            handleSelect={[Function]}
            meals={Array []}
            selectedMeals={Array []}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "Change",
            "handleClick": [Function],
            "ofTheDay": Object {
              "description": "",
              "meals": Array [],
              "name": "",
            },
            "openModal": [Function],
            "previewImage": "https://images.unsplash.com/photo-1482049016688-2d3e1b311543?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=c75e0437e819afdceeb3050a6bcdd71b&auto=format&fit=crop&w=653&q=80",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "closeModal": [Function],
            "handleSelect": [Function],
            "isVisible": undefined,
            "meals": Array [],
            "selectedMeals": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WorkStation Components RenderMeals should render correctly on shallow render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    handleSubmit={[Function]}
    meals={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="announce-item item-menu-options"
        >
          <p
            className="announce-item-key motd-main"
          >
             Meal Options 
          </p>
          <p
            className="announce-item-description motd-description"
          >
            Here you get to manage your meals, Add some and delete as you please. Remember to save some for later :).
          </p>
        </div>,
        <div
          className="menu-options-comp-two"
        >
          <div
            className="menu-options-grid list-grid"
          />
          <Unknown />
          <div
            className="edit-button"
            id="show-menu"
            onClick={[Function]}
          >
            New meal
          </div>
        </div>,
      ],
      "className": "menu-options-def",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="announce-item-key motd-main"
            >
               Meal Options 
            </p>,
            <p
              className="announce-item-description motd-description"
            >
              Here you get to manage your meals, Add some and delete as you please. Remember to save some for later :).
            </p>,
          ],
          "className": "announce-item item-menu-options",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Meal Options ",
              "className": "announce-item-key motd-main",
            },
            "ref": null,
            "rendered": " Meal Options ",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Here you get to manage your meals, Add some and delete as you please. Remember to save some for later :).",
              "className": "announce-item-description motd-description",
            },
            "ref": null,
            "rendered": "Here you get to manage your meals, Add some and delete as you please. Remember to save some for later :).",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="menu-options-grid list-grid"
            />,
            <Unknown />,
            <div
              className="edit-button"
              id="show-menu"
              onClick={[Function]}
            >
              New meal
            </div>,
          ],
          "className": "menu-options-comp-two",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
              "className": "menu-options-grid list-grid",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New meal",
              "className": "edit-button",
              "id": "show-menu",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "New meal",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="announce-item item-menu-options"
          >
            <p
              className="announce-item-key motd-main"
            >
               Meal Options 
            </p>
            <p
              className="announce-item-description motd-description"
            >
              Here you get to manage your meals, Add some and delete as you please. Remember to save some for later :).
            </p>
          </div>,
          <div
            className="menu-options-comp-two"
          >
            <div
              className="menu-options-grid list-grid"
            />
            <Unknown />
            <div
              className="edit-button"
              id="show-menu"
              onClick={[Function]}
            >
              New meal
            </div>
          </div>,
        ],
        "className": "menu-options-def",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="announce-item-key motd-main"
              >
                 Meal Options 
              </p>,
              <p
                className="announce-item-description motd-description"
              >
                Here you get to manage your meals, Add some and delete as you please. Remember to save some for later :).
              </p>,
            ],
            "className": "announce-item item-menu-options",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Meal Options ",
                "className": "announce-item-key motd-main",
              },
              "ref": null,
              "rendered": " Meal Options ",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Here you get to manage your meals, Add some and delete as you please. Remember to save some for later :).",
                "className": "announce-item-description motd-description",
              },
              "ref": null,
              "rendered": "Here you get to manage your meals, Add some and delete as you please. Remember to save some for later :).",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="menu-options-grid list-grid"
              />,
              <Unknown />,
              <div
                className="edit-button"
                id="show-menu"
                onClick={[Function]}
              >
                New meal
              </div>,
            ],
            "className": "menu-options-comp-two",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "menu-options-grid list-grid",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New meal",
                "className": "edit-button",
                "id": "show-menu",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "New meal",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WorkStation Components RenderOrdersComponent should render correctly on shallow render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="announce-item item-order-history"
        >
          <p
            className="announce-item-key order-history-heading"
          >
             Order history 
          </p>
          <p
            className="announce-item-description order-history-description"
          >
            Here are some insights on how you've fared today, it updates real time, so best believe we've got you! Enjoy
          </p>
        </div>,
        <div
          className="direct-to-order-page"
        >
          <p
            className="main-order-text"
          >
            Hey! so we built a whole new page for you to explore your orders on the fly. You can switch roles while in that page so you can see your orders as a regular user or as a caterer. Each comes with a different suite of capabilities. Enjoy!
            <br />
            <span
              className="order-history"
            >
               Visit order history
            </span>
          </p>
        </div>,
      ],
      "className": "direct-order-history",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="announce-item-key order-history-heading"
            >
               Order history 
            </p>,
            <p
              className="announce-item-description order-history-description"
            >
              Here are some insights on how you've fared today, it updates real time, so best believe we've got you! Enjoy
            </p>,
          ],
          "className": "announce-item item-order-history",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Order history ",
              "className": "announce-item-key order-history-heading",
            },
            "ref": null,
            "rendered": " Order history ",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Here are some insights on how you've fared today, it updates real time, so best believe we've got you! Enjoy",
              "className": "announce-item-description order-history-description",
            },
            "ref": null,
            "rendered": "Here are some insights on how you've fared today, it updates real time, so best believe we've got you! Enjoy",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="main-order-text"
          >
            Hey! so we built a whole new page for you to explore your orders on the fly. You can switch roles while in that page so you can see your orders as a regular user or as a caterer. Each comes with a different suite of capabilities. Enjoy!
            <br />
            <span
              className="order-history"
            >
               Visit order history
            </span>
          </p>,
          "className": "direct-to-order-page",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hey! so we built a whole new page for you to explore your orders on the fly. You can switch roles while in that page so you can see your orders as a regular user or as a caterer. Each comes with a different suite of capabilities. Enjoy!",
              <br />,
              <span
                className="order-history"
              >
                 Visit order history
              </span>,
            ],
            "className": "main-order-text",
          },
          "ref": null,
          "rendered": Array [
            "Hey! so we built a whole new page for you to explore your orders on the fly. You can switch roles while in that page so you can see your orders as a regular user or as a caterer. Each comes with a different suite of capabilities. Enjoy!",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Visit order history",
                "className": "order-history",
              },
              "ref": null,
              "rendered": " Visit order history",
              "type": "span",
            },
          ],
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="announce-item item-order-history"
          >
            <p
              className="announce-item-key order-history-heading"
            >
               Order history 
            </p>
            <p
              className="announce-item-description order-history-description"
            >
              Here are some insights on how you've fared today, it updates real time, so best believe we've got you! Enjoy
            </p>
          </div>,
          <div
            className="direct-to-order-page"
          >
            <p
              className="main-order-text"
            >
              Hey! so we built a whole new page for you to explore your orders on the fly. You can switch roles while in that page so you can see your orders as a regular user or as a caterer. Each comes with a different suite of capabilities. Enjoy!
              <br />
              <span
                className="order-history"
              >
                 Visit order history
              </span>
            </p>
          </div>,
        ],
        "className": "direct-order-history",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="announce-item-key order-history-heading"
              >
                 Order history 
              </p>,
              <p
                className="announce-item-description order-history-description"
              >
                Here are some insights on how you've fared today, it updates real time, so best believe we've got you! Enjoy
              </p>,
            ],
            "className": "announce-item item-order-history",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Order history ",
                "className": "announce-item-key order-history-heading",
              },
              "ref": null,
              "rendered": " Order history ",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Here are some insights on how you've fared today, it updates real time, so best believe we've got you! Enjoy",
                "className": "announce-item-description order-history-description",
              },
              "ref": null,
              "rendered": "Here are some insights on how you've fared today, it updates real time, so best believe we've got you! Enjoy",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="main-order-text"
            >
              Hey! so we built a whole new page for you to explore your orders on the fly. You can switch roles while in that page so you can see your orders as a regular user or as a caterer. Each comes with a different suite of capabilities. Enjoy!
              <br />
              <span
                className="order-history"
              >
                 Visit order history
              </span>
            </p>,
            "className": "direct-to-order-page",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Hey! so we built a whole new page for you to explore your orders on the fly. You can switch roles while in that page so you can see your orders as a regular user or as a caterer. Each comes with a different suite of capabilities. Enjoy!",
                <br />,
                <span
                  className="order-history"
                >
                   Visit order history
                </span>,
              ],
              "className": "main-order-text",
            },
            "ref": null,
            "rendered": Array [
              "Hey! so we built a whole new page for you to explore your orders on the fly. You can switch roles while in that page so you can see your orders as a regular user or as a caterer. Each comes with a different suite of capabilities. Enjoy!",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Visit order history",
                  "className": "order-history",
                },
                "ref": null,
                "rendered": " Visit order history",
                "type": "span",
              },
            ],
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WorkStation Components User Bio should render correctly on shallow render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    handleChange={[Function]}
    handleSubmit={[Function]}
    user={
      Object {
        "bio": "",
        "email": "",
        "firstname": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RenderUserBio
          user={
            Object {
              "bio": "",
              "email": "",
              "firstname": "",
            }
          }
        />,
        <SetUpNewKitchen
          handleChange={[Function]}
          handleSubmit={[Function]}
        />,
        undefined,
      ],
      "className": "show-user-and-kitchen-profile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "user": Object {
            "bio": "",
            "email": "",
            "firstname": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleChange": [Function],
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RenderUserBio
            user={
              Object {
                "bio": "",
                "email": "",
                "firstname": "",
              }
            }
          />,
          <SetUpNewKitchen
            handleChange={[Function]}
            handleSubmit={[Function]}
          />,
          undefined,
        ],
        "className": "show-user-and-kitchen-profile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "user": Object {
              "bio": "",
              "email": "",
              "firstname": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleChange": [Function],
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WorkStation Components WorkStation Main should render correctly on shallow render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WorkStation
    handleSubmit={[Function]}
    kitchen={
      Object {
        "bio": "",
        "description": "",
        "name": "",
      }
    }
    meals={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="announce-workstation"
        >
          <p
            className="announce-workstation-title"
          >
             Workstation 
          </p>
          <p
            className="announce-workstation-desc"
          >
             Control center for 
            
          </p>
        </div>,
        <div
          className="main-workstation-main"
        >
          <Unknown />
          <Connect(MenuOfTheDayContainer) />
          <Unknown
            handleSubmit={[Function]}
            meals={Array []}
          />
          <Unknown />
        </div>,
      ],
      "className": "main-workstation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="announce-workstation-title"
            >
               Workstation 
            </p>,
            <p
              className="announce-workstation-desc"
            >
               Control center for 
              
            </p>,
          ],
          "className": "announce-workstation",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Workstation ",
              "className": "announce-workstation-title",
            },
            "ref": null,
            "rendered": " Workstation ",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Control center for ",
                "",
              ],
              "className": "announce-workstation-desc",
            },
            "ref": null,
            "rendered": Array [
              " Control center for ",
              "",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown />,
            <Connect(MenuOfTheDayContainer) />,
            <Unknown
              handleSubmit={[Function]}
              meals={Array []}
            />,
            <Unknown />,
          ],
          "className": "main-workstation-main",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleSubmit": [Function],
              "meals": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="announce-workstation"
          >
            <p
              className="announce-workstation-title"
            >
               Workstation 
            </p>
            <p
              className="announce-workstation-desc"
            >
               Control center for 
              
            </p>
          </div>,
          <div
            className="main-workstation-main"
          >
            <Unknown />
            <Connect(MenuOfTheDayContainer) />
            <Unknown
              handleSubmit={[Function]}
              meals={Array []}
            />
            <Unknown />
          </div>,
        ],
        "className": "main-workstation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="announce-workstation-title"
              >
                 Workstation 
              </p>,
              <p
                className="announce-workstation-desc"
              >
                 Control center for 
                
              </p>,
            ],
            "className": "announce-workstation",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Workstation ",
                "className": "announce-workstation-title",
              },
              "ref": null,
              "rendered": " Workstation ",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Control center for ",
                  "",
                ],
                "className": "announce-workstation-desc",
              },
              "ref": null,
              "rendered": Array [
                " Control center for ",
                "",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown />,
              <Connect(MenuOfTheDayContainer) />,
              <Unknown
                handleSubmit={[Function]}
                meals={Array []}
              />,
              <Unknown />,
            ],
            "className": "main-workstation-main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleSubmit": [Function],
                "meals": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
